the-collab-lab/tcl-26-smart-shopping-list
# Show all item details

## Description



This adds the "Show all item details" button link to the top of the Shopping List in ListHeader.

![image](https://user-images.githubusercontent.com/61371242/131603192-b967917b-60aa-413b-87e0-342f4de7eae4.png)

Currently, this is a draft. I was able to get all of the items to open with a click of this link by moving the `showDetails` state variable up from ShoppingListItem to ListView and then passing it back down to both ListHeader and ShoppingList -> ShoppingListItem. However, I am not able to get an individual accordions to open. 

I started creating the ability to open a single accordion by passing the `showDetails` state down to the ShoppingListItem and setting `showDetails` as the initial state for a new state variable `showSingleDetail`:

`const [showSingleDetail, setShowSingleDetail] = useState(showDetails);`

But adding the following to the individual accordion icon is still not opening the individual panel.

`onClick={() => setShowSingleDetail(!showSingleDetail)}`

I've been using this as an example: https://codesandbox.io/s/festive-moore-cbqsn?file=/src/index.js

Also note that this branch includes the latest from the following branches as of Tuesday night, 8/31: `main`, `list-view`, and `details-accordion`

So, my real changes are only in:

- ListView.js
- ListHeader.js
- ShoppingList.js
- ShoppingListItem.js

You can ignore the rest of the changes. They are from bringing the other branches.

## Related Issue



Closes #82 : Add "Show all details" link

## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before



![image](https://user-images.githubusercontent.com/61371242/131603387-d5561b76-f765-4046-9511-519a2136f78e.png)

### After



![image](https://user-images.githubusercontent.com/61371242/131603407-5e580ce4-cfa2-4db2-a615-4c911f947fbe.png)


## Testing Steps / QA Criteria



1. git pull
2. Click "Show all item details" - all item accordions expand.
3. In progress - clicking on individual accordions should expand only the one accordion. This is still in progress.
